app/controllers/api/v1/search_controller.rb

```diff
@@ -0,0 +1,293 @@
+            key :required, true
+            key :type, :string
+            key :enum, ['study', 'cell']
+          end
+          parameter do
+            key :name, :facets
+            key :in, :query
+            key :description, 'User-supplied list facets and filters, formatted as: "facet_id:filter_value+facet_id_2:filter_value_2,filter_value_3"'
+            key :required, false
+            key :type, :string
+          end
+          parameter do
+            key :name, :terms
+            key :in, :query
+            key :description, 'User-supplied query string'
+            key :required, false
+            key :type, :string
+          end
+          response 200 do
+            key :description, 'Search parameters, Studies and StudyFiles'
+            schema do
+              key :title, 'Search Results'
+              property :type do
+                key :type, :string
+                key :description, 'Type of search performed'
+              end
+              property :terms do
+                key :type, :string
+                key :title, 'Keywords used in search'
+              end
+              property :page do
+                key :type, :string
+                key :title, 'Pagination control'
+              end
+              property :facets do
+                key :type, :array
+                key :title, 'SearchFacets'
+                key :description, 'Array of facets/filters used in search'
+                items do
+                  key :type, :object
+                  key :title, 'SearchFacet'
+                  property :id do
+                    key :type, :string
+                    key :description, 'ID of facet'
+                  end
+                  property :filters do
+                    key :type, :array
+                    key :description, 'Matching filters'
+                    items do
+                      key :type, :object
+                      key :title, 'Filter'
+                      property :name do
+                        key :type, :string
+                        key :description, 'Display value of filter'
+                      end
+                      property :id do
+                        key :type, :string
+                        key :description, 'ID value of filter'
+                      end
+                    end
+                  end
+                end
+              end
+              property :studies do
+                key :type, :array
+                items do
+                  key :title, 'Study, StudyFiles'
+                  key :'$ref', :SiteStudyWithFiles
+                end
+              end
+            end
+          end
+          response 406 do
+            key :description, 'Accept or Content-Type headers missing or misconfigured'
+          end
+          response 500 do
+            key :description, 'Server error or malformed query'
+          end
+        end
+      end
+
```

Comment:
this method should only be responsible for sanitizing the params and  identifiying the current user (if there is one), and returning the result. All actual search logic should live somewhere else.  Either in a static method on this controller or the study model

Reply:
cdf206c